lib/index.js

```diff
@@ -145,7 +145,7 @@ function parse(val, options, normalized) {
         }
     }
 
-    if (opts.float === true) {
+    if (opts.float === true && val.length) {
```

Comment:
Can we be a bit more explicit here, `val.length > 0`?

Reply:
Done.